### ISSUE TITLE
Use EfSecondLevelCache.Core in apps other than asp.net core apps

### ISSUE THREAD

**Issue by ysmoradi** [VahidN/EFSecondLevelCache.Core#31]
# Summary of the issue
I tried to use this library with Microsoft.EntityFrameworkCore.SQLite 2.0 in Xamarin based mobile/desktop app. (A .NET Standard 2.0 based project). I stopped with the [following exception](https://github.com/VahidN/EFSecondLevelCache.Core/blob/465f79feeada25f378939022a888e1f9af154076/src/EFSecondLevelCache.Core/EFCachedQueryExtensions.cs#L342):

Please add `AddEFSecondLevelCache` method to `IServiceCollection` and also add `UseEFSecondLevelCache` method to `IApplicationBuilder`.

## Environment

```
The in-use version:
Operating system: UWP 17763 | Android 9 | iOS 12
IDE: VS 2017 15.9
Version: EFSecondLevelCache.Core 1.7.1
```

I think it's not possible to use this library in other scnarios too. For example, if I write a .NET Core 2 console app which gets executed by azure jobs, I'll receive the same exception, because there is no asp.net core there.
I'm open to send a PR for this if you think it helps.
Thnaks in advance.

**Comment by VahidN** [comment]
`app.UseEFSecondLevelCache` is equal to the `EFServiceProvider.ApplicationServices = serviceProvider;`.
[More info from the unit tests project](https://github.com/VahidN/EFSecondLevelCache.Core/blob/master/src/Tests/EFSecondLevelCache.Core.Tests/TestsBase.cs#L64) which its setup is the same as console apps.

**Comment by ysmoradi** [comment]
This approach is not possible anymore due recent breaking changes

**Comment by VahidN** [comment]
Yes. Just use `services.AddEFSecondLevelCache();`. That would be enough.
